include/oneapi/tbb/concurrent_queue.h

```diff
@@ -131,8 +134,54 @@ class concurrent_queue {
         queue_allocator_traits::destroy(my_allocator, my_queue_representation);
         r1::cache_aligned_deallocate(my_queue_representation);
     }
+  
+    concurrent_queue& operator=( const concurrent_queue& other ){
+        //TODO: implement support for std::allocator_traits::propogate_on_container_copy_assignment
+        if (my_queue_representation != other.my_queue_representation) {
+            clear();
+	    my_queue_representation->assign(*other.my_queue_representation, my_allocator, copy_construct_item);
+	}
+	return *this;
+    }
+
+    concurrent_queue& operator=( concurrent_queue&& other ){
+        //TODO: implement support for std::allocator_traits::propogate_on_container_move_assignment
+        if (my_allocator == other.my_allocator){
+	    clear();
```

Comment:
As I see, this `clear()` call is not in both `if` and `else` branches - lets move it into the common part before the `if`.
Also, we need to add self-assignment check (either `if (this != &other)` or as it is done for copy assignment).